ragna/deploy/_ui/components/metadata_filters_builder.py

```diff
@@ -0,0 +1,307 @@
+
+    def __panel__(self):
+        return pn.Row(
+            self.key_select,
+            self.operator_select,
+            self.value_text_input,
+            self.delete_button,
+        )
+
+    def is_empty(self):
+        return self.key == "" and self.operator == "" and self.value == ""
+
+    def validate_value(self, value):
+        if value == "":
+            return False
+
+        if self.operator == "in" or self.operator == "not in":
+            values = value.split(",")
+            if len(values) > 1:
+                for val in values:
+                    if not self.validate_value(val.strip()):
+                        return False
+                return True
+            else:
+                value = values[0]
+
+        if METADATA_FILTERS[self.key]["type"] == int:
+            try:
+                int(value)
+            except ValueError:
+                return False
+        elif METADATA_FILTERS[self.key]["type"] == datetime:
+            try:
+                datetime.strptime(value, "%Y-%m-%d %H:%M:%S")
+            except ValueError:
+                return False
+
+        return True
+
+    def validate(self):
+        def change_valid_status(valid, widget):
+            if valid:
+                widget.css_classes = [
+                    c for c in widget.css_classes if c != "metadata-filter-error"
+                ]
+            else:
+                widget.css_classes = widget.css_classes + ["metadata-filter-error"]
+
+        if self.key == "":
+            change_valid_status(False, self.key_select)
+            return False
+
+        change_valid_status(True, self.key_select)
+
+        result = True
+        if self.operator == "":
+            change_valid_status(False, self.operator_select)
+            result = False
+        else:
+            change_valid_status(True, self.operator_select)
+
+        if not self.validate_value(self.value):
+            change_valid_status(False, self.value_text_input)
+            result = False
+        else:
+            change_valid_status(True, self.value_text_input)
+
+        return result
+
+    def __str__(self):
+        return f"{self.key} {self.operator} {self.value}"
+
+    def __repr__(self):
+        return f"[key:{self.key} \t op:{self.operator} \t val:{self.value}]"
+
+    def convert_operator(self, operator):
+        if operator == "==":
+            return MetadataFilter.eq
+        if operator == "!=":
+            return MetadataFilter.ne
+        if operator == ">":
+            return MetadataFilter.gt
+        if operator == "<":
+            return MetadataFilter.lt
+        if operator == ">=":
+            return MetadataFilter.ge
+        if operator == "<=":
+            return MetadataFilter.le
+        if operator == "in":
+            return MetadataFilter.in_
+        if operator == "not in":
+            return MetadataFilter.not_in
+
+        # if operator == "raw":
+        #    return MetadataFilter.RAW
```

Comment:
Let's avoid a switch yard here

```suggestion
        return {
            "==": MetadataOperator.EQ,
            "!=": MetadataOperator.NE,
            ...,
        }[operator]
```